pulsar/impl_partition_producer.go

```diff
@@ -293,14 +293,17 @@ func (p *partitionProducer) internalFlushCurrentBatch() {
 func (p *partitionProducer) internalFlush(fr *flushRequest) {
 	p.internalFlushCurrentBatch()
 
-	pi := p.pendingQueue.PeekLast().(*pendingItem)
-	pi.sendRequests = append(pi.sendRequests, &sendRequest{
-		msg: nil,
-		callback: func(id MessageID, message *ProducerMessage, e error) {
-			fr.err = e
-			fr.waitGroup.Done()
-		},
-	})
+	pi, ok := p.pendingQueue.PeekLast().(*pendingItem)
+	if ok {
+		pi.sendRequests = append(pi.sendRequests, &sendRequest{
+			msg: nil,
+			callback: func(id MessageID, message *ProducerMessage, e error) {
+				fr.err = e
+				fr.waitGroup.Done()
+			},
+		})
+	}
+
```

Comment:
If not ok, `fr.waitGroup.Done()` will not be executed, `wg.wait()` in send will not be released, and sending will be blocked indefinitely